feat: Added Music Island sharing support for Xiaomi's Hyper OS

ghhccghk opened the pull request:
A new utility class `IsLandHelp.kt` was introduced to build music sharing data for Xiaomi's Super Island.

In `GramophonePlaybackService`:
- When media metadata (song information) changes, `IsLandHelp.isLandMusicShare` is called to generate the corresponding `Bundle`.

In `MeiZuLyricsMediaNotificationProvider`:
- Modified the constructor to accept a `bundleProvider`.
- Added this `Bundle` to the notification's `extras` to enable Super Island functionality.

ghhccghk committed:
feat: Added Music Island sharing support for Xiaomi's Hyper OS

A new utility class `IsLandHelp.kt` was introduced to build music sharing data for Xiaomi's Super Island.

In `GramophonePlaybackService`:
- When media metadata (song information) changes, `IsLandHelp.isLandMusicShare` is called to generate the corresponding `Bundle`.

In `MeiZuLyricsMediaNotificationProvider`:
- Modified the constructor to accept a `bundleProvider`.
- Added this `Bundle` to the notification's `extras` to enable Super Island functionality.

Signed-off-by: ghhccghk <2137610394@qq.com>

nift4 commented on app/src/main/java/org/akanework/gramophone/logic/GramophonePlaybackService.kt:
```diff
@@ -599,6 +599,21 @@ class GramophonePlaybackService : MediaLibraryService(), MediaSessionService.Lis
 	    return super.onStartCommand(intent, flags, startId)
     }
 
+    override fun onMediaMetadataChanged(metadata: MediaMetadata) {
```
inside MeiZuLyricsMediaNotificationProvider.kt, addNotificationActions, the parameter "session" has player and you can access with "session.player.mediaMetadata". Can you move Bundle generation there? Thanks!

ghhccghk replied:
Let me see.

ghhccghk committed:
feat(notification): Directly retrieve metadata within the notification provider

Migrate metadata retrieval logic from `GramophonePlaybackService` to `MeiZuLyricsMediaNotificationProvider`.

This simplifies the service implementation and ensures notification construction can access required data more directly.

Signed-off-by: ghhccghk <2137610394@qq.com>

ghhccghk committed:
feat(i18n): Translate comments in IsLandHelp to English

Refactored IsLandHelp.kt by translating all KDoc and inline comments from Chinese to English for better international readability.

Also removed an unused `nfBundle` variable in `GramophonePlaybackService.kt`.

Signed-off-by: ghhccghk <2137610394@qq.com>

ghhccghk committed:
Partial modification reversal

Signed-off-by: ghhccghk <2137610394@qq.com>
